sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -178,15 +179,25 @@ static bool _on_link(const ipv6_addr_t *dst, unsigned *iface)
     }
 #endif  /* CONFIG_GNRC_IPV6_NIB_6LN */
     while ((entry = _nib_offl_iter(entry))) {
-        if ((entry->mode & _PL) && (entry->flags & _PFX_ON_LINK) &&
-            (ipv6_addr_match_prefix(dst, &entry->pfx) >= entry->pfx_len) &&
```

Comment:
Maybe the problem is that this should rather be a `==` or `<=` and that is why you get the wrong prefixes reported as off-link?